pennylane/beta/plugins/numpy_ops.py

```diff
@@ -0,0 +1,274 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Utility functions and numerical implementations of operations for
+the default.tensor device.
+"""
+
+import numpy as np
+from numpy.linalg import eigh
+
+# ========================================================
+#  utilities
+# ========================================================
+
+def spectral_decomposition(A):
+    r"""Spectral decomposition of a Hermitian matrix.
+
+    Args:
+        A (array): Hermitian matrix
+
+    Returns:
+        (vector[float], list[array[complex]]): (a, P): eigenvalues and hermitian projectors
+            such that :math:`A = \sum_k a_k P_k`.
+    """
+    d, v = eigh(A)
+    P = []
+    for k in range(d.shape[0]):
+        temp = v[:, k]
+        P.append(np.outer(temp, temp.conj()))
+    return d, P
+
+
+# ========================================================
+#  fixed gates/observables
+# ========================================================
+
+I = np.eye(2)
```

Comment:
(FYI: from some benchmarks, it seemed that having this as `np.array([[1, 0], [0, 1]])` is faster if called thousands of times. However, here it wouldn't make any difference, as `I` will be imported anyways and therefore `np.eye(2)` gets executed only once.)

Reply:
Nice info to keep in mind